field/TinyLang/Field/Printer.hs

```diff
@@ -17,11 +18,27 @@ import qualified Data.Vector               as Vector
 -- PrintStyle type determines whether printed variable names include
 -- these or not ("x_5" versus "x").  If we're going to re-parse the
 -- output of toString we probably don't want the IDs.
-data PrintStyle = WithIDs | NoIDs
+data PrintStyle = WithIDs | NoIDs | Prefix
+
+toStringSomeUniVar :: PrintStyle -> SomeUniVar f -> String
+toStringSomeUniVar ps = forget (toStringUniVar ps)
+
+toStringUniVar :: PrintStyle -> UniVar f a -> String
+toStringUniVar ps (UniVar uni var) = suffixOrPrefix ps uni $ toStringVar ps var where
+    suffixOrPrefix Prefix  = prefix
+    suffixOrPrefix _       = suffix
+    suffix :: Uni f a -> String -> String
+    suffix Bool   = (++ " : bool")
+    suffix Field  = (++ " : field")
+    suffix Vector = (++ " : vector")
+    prefix :: Uni f a -> String -> String
+    prefix Bool   = ("?" ++)
+    prefix Field  = ("$" ++)
+    prefix Vector = ("@" ++)
```

Comment:
Do we need prefixes for some reason?

Reply:
Not really.  I was thinking about making the syntax of `Raw` different than the pretty print of `Typed`, but you mentioned in a different issue that you prefer them to be the same, so I will change them back.